lib/tapioca/executor.rb

```diff
@@ -38,11 +44,29 @@ def run_in_parallel(&block)
       workers = (0...@number_of_workers).map do
         items = @queue.shift(@items_per_worker)
 
-        fork { block.call(items.shift) until items.empty? }
+        fork do
+          read.close
+          result = items.map { |item| block.call(item) }.compact
+
+          packed = [Marshal.dump(result)].pack("m")
+          write.puts("-#{packed}") unless result.empty?
```

Comment:
Can we extract the delimiter (`-` in this case) to be a constant somewhere so that we can use the same constant below when parsing? It would also be a convenient place to note that we chose `-` since it is not in the Base64 alphabet that we are using to serialize the data on the pipe.